spec/message.abnf

```diff
@@ -70,9 +70,10 @@ reserved-char  = %x00-08        ; omit HTAB and LF
                / %x7E-D7FF      ; omit surrogates
                / %xE000-10FFFF
 
-; based on https://www.w3.org/TR/xml/#NT-Name,
-; but cannot start with U+003A COLON ":"
-name = name-start *name-char
+; based on https://www.w3.org/TR/REC-xml-names/#NT-QName
+name       = [namespace ":"] name-body
+namespace  = name-part
+name-part  = name-start *name-char
 name-start = ALPHA / "_"
            / %xC0-D6 / %xD8-F6 / %xF8-2FF
```

Comment:
There is no definition for `name-body`, but this grammar is ambiguous if that is intended to be `name-part`—which ends in `*name-char` and therefore can itself include colons. For example, would parsing "foo:bar:baz" as a `name` be `{ namespace: "foo", name-part: "bar:baz" }` or `{ namespace: "foo:bar", name-part: "baz" }` or `{ name-part: "foo:bar:baz" }`?

Reply:
I agree that we need to fix the ambiguity here. I think #519 is the right place to do it, so that we have a chance to holistically look at the syntax and consider other productions. We don't need to block the PR.

My preference, however, would be to make this PR complete, i.e. fix the ambiguity *somehow* before merging. Two options for temporary solutions:

* Use `/` as a temporary namespace separator.
* Make `namespace` consist only of `name-start` characters.

Reply:
I see that the changes to `syntax.md` mention `NCName`. @aphillips do you intend to factor `NCName` into `message.abnf` in this PR?

Reply:
Removed colon (which I had intended in the first place 🤦🤷‍♂️🙈). As noted above, this makes us match XML names.